### PR TITLE
Bukkit enum eradication futureproofing

### PR DESCRIPTION
**Pull request by Warriorrrr**
#### Description: 

This PR makes some changes required for the upcoming removal of various Bukkit enums, most notably material & entitytype.

Since some methods were returning EnumSets, we do have some breaking changes unfortunately. Mainly in TownyWorld and TownySettings, I've changed some return types and parameter types to Collection to avoid breakage if we ever change the impl again.

Also fixes an issue with large fireballs left over from when we were doing entity types with class names.
____
- [x] I have tested this pull request for defects on a server. 

By making this pull request, I represent that I have the right to waive copyright and related rights to my contribution, and agree that all copyright and related rights in my contributions are waived, and I acknowledge that the TownyAdvanced organization has the copyright to use and modify my contribution under the Towny [License](https://github.com/LlmDl/Towny/blob/master/LICENSE.md) for perpetuity.

### CODE REVIEW

**Comment by Warriorrrr** [comment]
I forgot about switches, that will be fun

File: Towny/src/main/java/com/palmergames/bukkit/towny/event/executors/TownyActionEventExecutor.java [hunk 1]
```diff
@@ -184,12 +184,9 @@ private static boolean isNotCandle(Block block) {
 	}
 	
 	private static boolean isNotFireSpreadBypassMat(Block block) {
-		switch (block.getType()) {
-			case CAMPFIRE:
-				break;
-			default:
-				block = block.getRelative(BlockFace.DOWN);
-		}
+		if (ItemLists.CAMPFIRES.contains(block.getType()))
```
Comment (LlmDl) [inline Towny/src/main/java/com/palmergames/bukkit/towny/event/executors/TownyActionEventExecutor.java:187]: This is meant to do line 190 if it is a Campfire, it looks like you might have left out a `!`.

File: Towny/src/main/java/com/palmergames/bukkit/towny/listeners/TownyInventoryListener.java
Comment (LlmDl) [inline Towny/src/main/java/com/palmergames/bukkit/towny/listeners/TownyInventoryListener.java:1]: This change looks like it could be a separate PR, doesn't really seem to be related to enumset->set. You can leave it in I guess.

File: Towny/src/main/java/com/palmergames/bukkit/util/ItemLists.java [hunk 2]
```diff
@@ -266,106 +240,19 @@ public boolean contains(@NotNull ItemStack itemStack) {
...
-
-			for (String name : names) {
-				try {
-					exceptions.add(Material.valueOf(name.toUpperCase(Locale.ROOT)));
-				} catch (IllegalArgumentException ignored) {}
-			}
-
-			return this;
-		}
+	public static Builder<Material, ItemLists> newBuilder() {
```
Comment (LlmDl) [inline Towny/src/main/java/com/palmergames/bukkit/util/ItemLists.java:255]: Nice to see a public constructor, Goosius and I had just recently had an issue where we couldn't use ItemLists because PredicateLists couldn't be made with other plugins.

**Review by LlmDl — CHANGES_REQUESTED** [review]
Just one regression I think I spotted in TownyActionEventExecutor.

File: Towny/src/main/java/com/palmergames/bukkit/towny/listeners/TownyInventoryListener.java
Comment (Warriorrrr) [reply Towny/src/main/java/com/palmergames/bukkit/towny/listeners/TownyInventoryListener.java:1]: Switches are affected by the change too, it will compile for now (and still run since spigot has some bytecode rewrites in place to make it work), but once it's merged any switches with materials will no longer compile.

**Comment by LlmDl** [comment]
So we're not losing any backwards compatibility yet? When spigot does merge their new registry stuff, are we still able to support back to 1.16.5 and we'll only lose that when they drop their compatibility layer?

**Comment by Warriorrrr** [comment]
Going to see if this still works on older versions when compiling against a version of the API where the enums are replaced, would be nice to have this included in the next release since 1.20 is coming soon.

File: Towny/src/main/java/com/palmergames/bukkit/config/migration/WorldMigrationAction.java [hunk 3]
```diff
@@ -1,66 +1,73 @@
...
 					.collect(Collectors.toList()), change.key, change.value));
 		else
-			townyWorld.setUnclaimedZoneIgnore(splitMats(StringMgmt.join(townyWorld.getUnclaimedZoneIgnoreMaterials(), ",") + change.value));
+			townyWorld.setUnclaimedZoneIgnore(splitMats(townyWorld.getUnclaimedZoneIgnoreMaterials().stream().map(type -> type.getKey().toString()).collect(Collectors.joining(",")) + change.value));
 	})),
 	UPDATE_WORLD_EXPLOSION_REVERT_BLOCKS(((townyWorld, change) -> {
 		if (change.type == MigrationType.REPLACE)
 			townyWorld.setPlotManagementWildRevertMaterials(replaceAll(townyWorld.getPlotManagementWildRevertBlocks().stream()
-					.map(mat -> mat.name())
+					.map(mat -> mat.getKey().getKey())
```
Comment (LlmDl) [inline Towny/src/main/java/com/palmergames/bukkit/config/migration/WorldMigrationAction.java:48]: extract mat.getKey().getKey() to a getMatName(mat) method plz
